classes/01class/exercises/c01-git04/afaryy/README.md

```diff
@@ -0,0 +1,58 @@
+# c01-git04
+
+
+## Secrets (c01-git04)
+
+Perform the following commands:
+1. In the same repository of the previous exercise:
+2. Add a file called `my_env.txt` that contains
+    ```
+    SERVICE_NAME=account-management
+    ENVIRONMENT=prod
+    PASSWORD=pass1234
+    ```
+1. Commit it to your local repository
+2. Check the log with `git log`
+
+## Questions (answer inline)
+
+1. Let's suppose you remotely pushed the `my_env.txt` file above. A colleague asks you to remove this information from Git. What's your colleague worried about?  
+You should not include sensitive data like password in yout file.
+
+2. If you modify the file in your workspace, then commit and push it, will it be enough to erase this password information from the repository? (It's not). Why?  
+No, the file including the password info still exists in git history.
+
+3. If you delete the file and push it, then create a new one with the rest of the information, is it enough? (It's not). Why?  
+No, this also can not remove the git history of the file.
+4. How to fix this? How do you remove something from Git history when it is in the remote repository?   
+The "git filter-branch" works good to remove your file with sensitive data from the history of your repository.
+
```

Comment:
Even if you delete the file from the history locally you need to push it. You will need to force-push it since you changed history.

Do you think after force-pushing and removing that file from the remote repository we are safe again? What if someone already cloned that repository?

Your answer is not wrong, but would be nice to add a bit more of information regarding the security perspective.